vcd_cli/gateway.py

```diff
@@ -48,3 +54,131 @@ def list_gateways(ctx):
         stdout(result, ctx, show_id=False)
     except Exception as e:
         stderr(e, ctx)
+
+
+@gateway.command(short_help='show edge gateway details')
+@click.pass_context
+@click.argument('name',
+                metavar='<name>',
+                required=True)
+def info(ctx, name):
+    restore_session(ctx, vdc_required=True)
+    client = ctx.obj['client']
+    vdc_href = ctx.obj['profiles'].get('vdc_href')
+    vdc = VDC(client, href=vdc_href)
+    edge_gateways = vdc.list_edge_gateways()
+    for gw in edge_gateways:
+        if name == gw.get('name'):
+            resource = client.get_resource(gw.get('href'))
+            result = edge_gateway_to_dict(resource)
```

Comment:
edge_gateway_to_dict(..) basically retrieves only name and href. Info command should display much more than that.
Any specific reason why you had to use new method edge_gateway_to_dict()? 
Have you tried vcd_cli.utils.to_dict()?

fyi, these utility methods should go to utils.py file